### PR TITLE
Update .gitignore for Rider

### PR DESCRIPTION
**Pull request by alrz**

### CODE REVIEW

**Comment by Pilchie** [comment]
@dotnet/roslyn-infrastructure

**Comment by jasonmalinowski** [comment]
@dotnet-bot retest this please.

You happened to make the PR when we had a build break in the branch.

**Comment by alrz** [comment]
> All checks have failed.

That was scary.

File: .gitignore [hunk 1]
```diff
@@ -179,3 +179,8 @@ $RECYCLE.BIN/
 
 # Visual Studio Code
 .vscode/
+
+# JetBrains Rider
+.idea/
+*.DotSettings
```
Comment (sharwell) [inline .gitignore]: ❔ Should this go under the ReSharper settings above? I vaguely remember seeing people put it there in the past, but I could be wrong and/or things could have changed.

Comment (alrz) [inline .gitignore]: You're right, it does belong to ReSharper since R# in VS can also produce those files.

**Comment by alrz** [comment]
tagging @jasonmalinowski, this is all green.

**Pull request merged**